salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +53,100 @@ def node_info(node, ca_minion):
     return result
 
 
+def get_cluster_version(api_client):
+    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+    try:
+        namespace = coreV1.read_namespace(name="kube-system")
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to read namespace information {}'.format(exc)
+        )
+    annotations = namespace.metadata.annotations
+    annotation_key = 'metalk8s.scality.com/cluster-version'
+    if not annotations or annotation_key not in annotations:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return namespace.metadata.annotations[annotation_key]
+
+
+def iso_timestamp_converter(timestamp):
+    if not timestamp:
+        return None
+    return timestamp.isoformat()
+
+
+def get_storage_classes(api_client):
+    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
+    storage_classes = {}
+    try:
+        storageclass_list = storageV1.list_storage_class()
+    except (ApiException, HTTPError) as exc:
+        log.exception('Exception when calling storageV1->list_storage_class')
+        return None
+    for storageclass in storageclass_list.items:
+        # Need to convert the datetime object in storageclass to ISO format in
+        # order to make them serializable.
+        storageclass.metadata.creation_timestamp = iso_timestamp_converter(
+            storageclass.metadata.creation_timestamp
+        )
+        storageclass.metadata.deletion_timestamp = iso_timestamp_converter(
+            storageclass.metadata.deletion_timestamp
+        )
+        storage_classes[storageclass.metadata.name] = storageclass.to_dict()
+
+    return storage_classes
+
+
+def list_volumes(api_client, minion_id):
+    results = {}
+    customObjectApi = kubernetes.client.CustomObjectsApi(api_client=api_client)
+    storage_classes = get_storage_classes(api_client)
+
+    if not storage_classes:
+        return {}
+
+    try:
+        volumes = customObjectApi.list_cluster_custom_object(
+            group="storage.metalk8s.scality.com",
+            version="v1alpha1",
+            plural="volumes"
+        )
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to get volumes {}'.format(exc)
+        )
+
+    local_volumes = (
+        volume for volume in volumes['items']
+        if volume['spec']['nodeName'] == minion_id
+    )
+    for volume in local_volumes:
+        name = volume['metadata']['name']
+        storageclass = storage_classes.get(
+            volume['spec']['storageClassName']
+        )
+        if storageclass:
+            volume['spec']['storageClassName'] = storageclass
+            result = volume
+        else:
+            result = None
+            log.error('Invalid storageclass for {} volume'.format(name))
```

Comment:
Could be interesting to have the content of the `storageclass` variable if it's evaluated as `False`. Is it a None or something else?

Reply:
Actually, the error message should be:

```suggestion
            log.error('Missing StorageClass "{}" for Volume "{}"'.format(storageClassName, name))
```